### PR TITLE
Add prevent suicide to minds and add tag control commands

### PR DESCRIPTION
**Pull request by Chief-Engineer**
## About the PR 
Adds options to prevent suicide for minds and ~~mind components~~ adds commands to edit tags ingame.

**Media**

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

### CODE REVIEW

**Comment by Elijahrane** [comment]
Isn't there already a CannotSuicide tag? Look at mice

File: Content.Server/Administration/Commands/TagCommands.cs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    }
+
+    [AdminCommand(AdminFlags.Debug)]
+    public sealed class RemoveTagCommand : LocalizedCommands
+    {
+        [Dependency] private readonly IEntityManager _entityManager = default!;
+
+        public override string Command => "removetag";
+        public override string Description => "Removes a tag from a given entity";
+        public override string Help => "Usage: removetag <entity uid> <tag>";
```
Comment (mirrorcult) [inline Content.Server/Administration/Commands/TagCommands.cs:51]: completions would be really nice

Comment (mirrorcult) [reply Content.Server/Administration/Commands/TagCommands.cs:51]: with the current tagcomp tags for removetag, and `CompletionHelper.PrototypeIDs<TagPrototype>()` for addtag

**Review by mirrorcult — CHANGES_REQUESTED** [review]

File: Content.Server/Administration/Commands/TagCommands.cs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+{
+    [AdminCommand(AdminFlags.Debug)]
+    public sealed class AddTagCommand : LocalizedCommands
+    {
+        [Dependency] private readonly IEntityManager _entityManager = default!;
+
+        public override string Command => "addtag";
+        public override string Description => "Adds a tag to a given entity";
+        public override string Help => "Usage: addtag <entity uid> <tag>";
+
```
Comment (metalgearsloth) [inline Content.Server/Administration/Commands/TagCommands.cs:15]: This needs localizing.

File: Content.Server/Administration/Commands/TagCommands.cs [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+                return;
+            _entityManager.EnsureComponent<TagComponent>(entityUid);
+
+            if (tagSystem.TryAddTag(entityUid, args[1]))
+            {
+                shell.WriteLine($@"Added {args[1]} to {entityUid}.");
+            }
+            else
+            {
+                shell.WriteError($@"Could not add {args[1]} to {entityUid}.");
```
Comment (metalgearsloth) [inline Content.Server/Administration/Commands/TagCommands.cs:40]: Same herre, also why the @

File: Content.Server/Administration/Commands/TagCommands.cs [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+            }
+
+            if (args.Length == 2&& EntityUid.TryParse(args[0], out var entityUid) && _entityManager.TryGetComponent(entityUid, out TagComponent? tagComponent))
+            {
+                return CompletionResult.FromHintOptions(tagComponent.Tags,
+                    Loc.GetString("tag-command-arg-tag"));
+            }
+
+            return CompletionResult.Empty;
+        }
```
Comment (metalgearsloth) [inline Content.Server/Administration/Commands/TagCommands.cs:111]: As above

File: Content.Server/Mind/MindSystem.cs [hunk 5]
```diff
@@ -154,4 +156,15 @@ private void OnExamined(EntityUid uid, MindComponent mind, ExaminedEvent args)
...
+    private void OnSuicide(EntityUid uid, MindComponent component, SuicideEvent args)
+    {
+        if (args.Handled)
+            return;
+
+        if (component.HasMind && component.Mind!.PreventSuicide)
+        {
+            args.BlockSuicideAttempt(true);
+        }
+    }
```
Comment (metalgearsloth) [inline Content.Server/Mind/MindSystem.cs:169]: Shouldn't this be its own component.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

**Review by DrSmugleaf — APPROVED** [review]

**Pull request merged**